core/src/main/java/org/gbif/pipelines/core/functions/interpretation/CountryCodeInterpreter.java

```diff
@@ -0,0 +1,39 @@
+package org.gbif.pipelines.core.functions.interpretation;
+
+import org.gbif.api.vocabulary.Country;
+import org.gbif.common.parsers.CountryParser;
+import org.gbif.common.parsers.core.ParseResult;
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueType;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.core.functions.interpretation.error.LineageType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CountryCodeInterpreter implements Interpretable<String, String> {
+
+  @Override
+  public String interpret(String input) throws InterpretationException {
+    List<Issue> issues = new ArrayList<>();
+    List<Lineage> lineages = new ArrayList<>();
+
+    ParseResult<Country> parseCountry = CountryParser.getInstance().parse(input.trim());
+    if (parseCountry.isSuccessful()) {
+      return parseCountry.getPayload().getIso3LetterCode();
+    } else {
+      issues.add(Issue.newBuilder()
+                   .setIssueType(IssueType.PARSE_ERROR)
+                   .setRemark(parseCountry.getError().getMessage())
+                   .build());
+      lineages.add(Lineage.newBuilder()
+                     .setLineageType(LineageType.SET_TO_NULL)
+                     .setRemark("Since the parse on countryCode failed, interpreting as null")
+                     .build());
+      InterpretationException iex = new InterpretationException(issues, lineages);
+      iex.setInterpretedValue(null);
```

Comment:
InterpretationException class do not have a method to set isInterpretedObjectAvailable. It can only be read or set implicitly by a method setInterpretedValue.  

Why we need isInterpretedObjectAvailable ?
We need it because, there are 2 cases when an interpretedValue is null, once when it is not initialised, other case when we could not parse a raw value properly and interpret it as null. These 2 cases are totally different and should be identified, 1 st case infers that there was no value interpreted, whereas 2 nd case there was interpretation made. Hence the flag is set to distinguish these 2 cases. 
Not giving access/control to the user of api on the variable isInterpretedObjectAvailable is necessary to avoid corruption of the state of this object. As this state is very important to distinguish interpretation was done or not, it is important to not set the interpretedValue by user when not needed.